airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -5,29 +5,80 @@
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamsToReset = new LinkedList<>();
+  // TODO: Once we are sure that the legacy way of transmitting the state is not use anymore, we need to remove this variable and the associated
+  //  checks
+  private boolean isResetBasedForConfig;
+  private boolean isStarted = false;
+  private Optional<StateWrapper> stateWrapper;
 
   public EmptyAirbyteSource() {
     hasEmittedState = new AtomicBoolean();
   }
 
   @Override
-  public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) throws Exception {
-    // no op.
+  public void start(final WorkerSourceConfig workerSourceConfig, final Path jobRoot) throws Exception {
+
+    if (workerSourceConfig == null || workerSourceConfig.getSourceConnectionConfiguration() == null) {
+      // TODO: When the jobConfig is fully updated and tested, we can remove this extra check that makes
+      // us compatible with running a reset with
+      // a null config
+      isResetBasedForConfig = false;
+    } else {
+      final ResetSourceConfiguration resetSourceConfiguration;
+      resetSourceConfiguration = parseResetSourceConfigurationAndLogError(workerSourceConfig);
+      streamsToReset.addAll(resetSourceConfiguration.getStreamsToReset());
+
+      if (streamsToReset.isEmpty()) {
+        // TODO: This is done to be able to handle the transition period where we can have no stream being
+        // pass to the configuration because the
+        // logic of populating this list is not implemented
+        isResetBasedForConfig = false;
+      } else {
+        stateWrapper = StateMessageHelper.getTypedState(workerSourceConfig.getState().getState());
+
+        if (stateWrapper.isPresent() &&
+            stateWrapper.get().getStateType() == StateType.LEGACY &&
+            !isResetAllStreamsInCatalog(workerSourceConfig)) {
+          log.error("The state a legacy one but we are trying to do a partial update, this is not supported.");
+          throw new IllegalStateException("Try to perform a partial reset on a legacy state");
```

Comment:
Shouldn't we have the same check for `StateType.GLOBAL`?

Reply:
No, we can do a partial reset on a GLOBAL state (that the reason why we extracted it). What we can't do is to perform a partial sync.